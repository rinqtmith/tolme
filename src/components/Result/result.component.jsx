import React from 'react';

import { Message } from 'semantic-ui-react';

const ResultComponent = ({ result, checkValue, value }) => (
  <React.Fragment>
    {!result.includes('Value') ? (
      !result.includes('Thickness') ? (
        value !== 'angular' ? (
          <Message
            style={{ marginBottom: '1em' }}
          >{`${checkValue}±${result}`}</Message>
        ) : (
          <Message style={{ marginBottom: '1em' }}>{`±${result}`}</Message>
        )
      ) : (
        <Message style={{ marginBottom: '1em' }}>{result}</Message>
      )
    ) : (
      <Message style={{ marginBottom: '1em' }}>{result}</Message>
    )}
  </React.Fragment>
);

export default ResultComponent;
