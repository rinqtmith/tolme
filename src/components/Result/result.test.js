import React from 'react';
import { shallow } from 'enzyme';

import Result from './result.component';

describe('Result Component', () => {
  it("should render component when result doesn't have value, and value is not angular", () => {
    const mockProps = {
      result: '0,5',
      checkValue: '12',
      value: 'radius',
    };

    const wrapper = shallow(<Result {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should render component when result doesn't have value, and value is angular", () => {
    const mockProps = {
      result: '0,5',
      checkValue: '12',
      value: 'angular',
    };

    const wrapper = shallow(<Result {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render component when result have value', () => {
    const mockProps = {
      result: 'Value must be lower than 4000',
      checkValue: '12',
      value: 'angular',
    };

    const wrapper = shallow(<Result {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render component when result have value', () => {
    const mockProps = {
      result: 'Thickness should be greater than 0',
      checkValue: '12',
      value: 'angular',
    };

    const wrapper = shallow(<Result {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
