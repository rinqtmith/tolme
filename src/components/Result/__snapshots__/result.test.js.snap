// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Component should render component when result doesn't have value, and value is angular 1`] = `
<Fragment>
  <Message
    style={
      Object {
        "marginBottom": "1em",
      }
    }
  >
    ±0,5
  </Message>
</Fragment>
`;

exports[`Result Component should render component when result doesn't have value, and value is not angular 1`] = `
<Fragment>
  <Message
    style={
      Object {
        "marginBottom": "1em",
      }
    }
  >
    12±0,5
  </Message>
</Fragment>
`;

exports[`Result Component should render component when result have value 1`] = `
<Fragment>
  <Message
    style={
      Object {
        "marginBottom": "1em",
      }
    }
  >
    Value must be lower than 4000
  </Message>
</Fragment>
`;

exports[`Result Component should render component when result have value 2`] = `
<Fragment>
  <Message
    style={
      Object {
        "marginBottom": "1em",
      }
    }
  >
    Thickness should be greater than 0
  </Message>
</Fragment>
`;
